generated/nidaqmx/_grpc_interpreter.py

```diff
@@ -22,13 +22,42 @@ def __init__(self, grpc_options):
     def _invoke(self, func, request, metadata=None):
         try:
             response = func(request, metadata=metadata)
-            error_code = response.status
-            error_message = ''
-        #Todo: AB#2325876: The error handling logic will be updated.
         except grpc.RpcError as rpc_error:
-            raise rpc_error
+            error_code = -1
+            error_message = rpc_error.details()
+            samps_per_chan_read = None
+            samps_per_chan_written = None
+            for entry in rpc_error.trailing_metadata() or []:
+                if entry.key == 'ni-error':
+                    value = entry.value if isinstance(entry.value, str) else entry.value.decode('utf-8')
+                    try:
+                        error_code = int(value)
+                    except ValueError:
+                        error_message += f'\nError status: {value}'
+                if entry.key == "samps_per_chan_read":
+                    samps_per_chan_read = entry.value
+                if entry.key == "samps_per_chan_written":
+                    samps_per_chan_written = entry.value
+            grpc_error = rpc_error.code()
+            if grpc_error == grpc.StatusCode.UNAVAILABLE:
+                error_message = 'Failed to connect to server'
```

Comment:
Error code is used as -1 for the `gRPC` specific errors. Should this be handled differently?

Reply:
You can see how nimi-python handles this situation here: https://github.com/ni/nimi-python/blob/master/generated/nidcpower/nidcpower/_grpc_stub_interpreter.py.

The summary is that they created a new class of error called an [`RpcError`](https://github.com/ni/nimi-python/blob/ca585de99ca191703889eb88841efc4cc6f59b09/generated/nidcpower/nidcpower/errors.py#L46) that is different from a `DriverError` for situations like this. They initialize the code to `None` and raise the `RpcError` in that case. I think that is a pattern we should follow here.

Reply:
I have created an `RPCError` class in `errors.py` which will handle the errors specific to the gRPC server.